src/messaging.rs

```diff
@@ -1,18 +1,18 @@
 /// Message sent by a given source.
 #[derive(Clone, Debug)]
 pub struct SourcedMessage<M, N> {
+    /// The ID of the sender.
     pub source: N,
+    /// The message's content.
     pub message: M,
 }
 
-/// Message destination can be either of the two:
-///
-/// 1) `All`: all remote nodes.
-///
-/// 2) `Node(id)`: remote node `id`.
```

Comment:
I'd probably simplify this to just: "The intended message recipient(s)." It's pretty clear by looking at the variants what the possibilities are and it's a bit of a duplication.

Reply:
This comment applies to the entire doc comments.

Reply:
I might comment...